bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -846,22 +877,42 @@ public void stop() {
 	webView.Stop();
 }
 
+private void writeToDefaultPathDOM() {
+	if(html != null && URI.create(getUrl()).equals(Browser.BASE_URI)) {
+		boolean test = jsEnabled;
+		jsEnabled = true;
+		execute("document.open(); document.write(`" + html + "`); document.close();");
+		jsEnabled = test;
+		this.html = null;
+	}
+}
+
 @Override
 public boolean setText(String html, boolean trusted) {
-	char[] data = new char[html.length() + 1];
-	html.getChars(0, html.length(), data, 0);
-	return webView.NavigateToString(data) == COM.S_OK;
+	return setWebpageData(Browser.BASE_URI.toASCIIString(), null, null, html);
 }
 
-@Override
-public boolean setUrl(String url, String postData, String[] headers) {
+private boolean setWebpageData(final String url, String postData, String[] headers, String html) {
+	// If the create() method hasn't finished executing, queue the call to wait for the browser to finish initializing
+	if (!initializationProcess.isDone()) {
+		browser.getDisplay().asyncExec(() -> {
+			waitForInitialization();
+			setWebpageData(url, postData, headers, html);
+			browserResize(new Event());
+		});
+		return true;
+	}
 	// Feature in WebView2. Partial URLs like "www.example.com" are not accepted.
 	// Prepend the protocol if it's missing.
+	final String normalisedUrl;
```

Comment:
We need url to be final otherwise, it can't be used in the async call. we need a new variable anyways, one for the aync call and one for the method here. Can create a final var from url and pass it on to async call (thats the second option)

Reply:
In that case:
```java
private boolean setWebpageData(String url, String postData, String[] headers, String html) {
	// If the create() method hasn't finished executing, queue the call to wait for the browser to finish initializing
	if (!initializationProcess.isDone()) {
		final String fUrl = url;
		browser.getDisplay().asyncExec(() -> {
			waitForInitialization();
			setWebpageData(fUrl, postData, headers, html);
			browserResize(new Event());
		});
		return true;
	}
	// Feature in WebView2. Partial URLs like "www.example.com" are not accepted.
	// Prepend the protocol if it's missing.
	if (!url.matches("[a-z][a-z0-9+.-]*:.*")) {
		url = "http://" + url;
	}
	...
```

Reply:
Done